### DISCUSSION TITLE
WHDLOAD games launching with "wrong" config

### DISCUSSION THREAD

**Discussion by sittingduck77** [BlitterStudio/amiberry#1212] [Q&A] [answered]
I'm running Amiberry from within retropie. A lot of games seem to launch using A500 config when I would expect A1200 to work better - eg. Zeewolf 1 and 2. Is something wrong with my setup or is this by design? I have updated the Whdload XML file. Thanks.

**Comment by giantclambake** [comment]
Zeewolf 1 & 2 are setup for OCS/ECS Amiga machine types. so this is expected behavior here AFAICT.

**Comment by giantclambake** [comment]
I should elucidate some, as I just tested this with an A1200 machine config, and it does actually run better like that.... not at authentic & accurate speed, but better.

The best way to manage this, is to create a local config.uae file for the title(s) you wish to treat in this way, in the Configuration files PATH of your amiberry installation. Amiberry looks in this path location first, to see if there's a matching config for the loaded whdload.lha file, and if it exists it will use that to setup the emulation, not the XML file directives.  To do this...

Start Amiberry -> Quickstart -> select WHDLoad archive -> Start ...wait for game load and splashscreen/music --> F12 to GUI
Configurations -> Save (name/description fields will already be correct)
CPU & FPU -> select 68020
Chipset -> select AGA and set Chipset extra: to A1200
Configurations -> Save

Now any time you load the specific whdload game archive, it will start/run with an A1200 Amiga machine type.

HTH

[Zeewolf2_v5.00_0608.uae.gz](https://github.com/BlitterStudio/amiberry/files/13829783/Zeewolf2_v5.00_0608.uae.gz)

**Comment by thomasjr1** [comment]
Check Configuration Files:
Ensure that the configuration files for WHDLoad are correctly set up. Look for any typos or errors in the configuration settings.

Update WHDLoad:
Make sure you are using the latest version of WHDLoad. Developers often release updates to fix bugs and improve compatibility.

Game-Specific Configurations:
Some games may require specific configurations. Check if there are any recommended settings for the particular game you are trying to run.

Kickstart ROM Version:
Verify that you are using the correct Kickstart ROM version. Some games may require a specific version of the Amiga Kickstart ROM to run properly.

Memory Settings:
Adjust the memory settings in the WHDLoad configuration. Some games may have specific memory requirements, and configuring them incorrectly can lead to issues.

Readme Files:
Consult the readme files that come with the WHDLoad games. Developers often provide specific instructions and troubleshooting [tips](https://trafficridermod.app/traffic-rider-tips-and-tricks/).

Support Forums:
Visit Amiga or WHDLoad support forums. Other users may have experienced similar issues and could provide solutions or workarounds.

Emulator Settings:
If you are using an emulator to run WHDLoad games, check the emulator settings. Ensure that the emulator is configured correctly for WHDLoad compatibility.

Amiga Model:
Some games may have specific requirements based on the Amiga model. Check if the game you're trying to run has any such restrictions.

Reinstall WHDLoad:
If all else fails, try reinstalling WHDLoad. Sometimes, files may become corrupted, and a fresh installation can resolve issues.

**Reply by sittingduck77** [reply]
Thanks, I believe everything is in order. 
Thanks for elaborating, giantclambake. Yeah, I'll set up individual configs for these games - I guess my point is, that the Whdload XML-file should reflect that AGA setups are better for these games and run them that way pr. default. Who updates/decides the settings in the Amiberry XML file?

**Reply by midwan** [reply]
The XML is updated from this repo: https://github.com/HoraceAndTheSpider/Amiberry-XML-Builder
Feel free to open an issue there, about these games.

**Reply by sittingduck77** [reply]
Thanks - I'll have a look. And please know, that I don't think of this as "issues" - I appreciate all the effort put into Amiberry :-)

**Reply by HoraceAndTheSpider** [reply]
I am happy to include game specific changes for 'better' hardware on a case-by-case basis where it is clearly demonstrated the enhanced setup is beneficial.

However, I prefer the 'minimum' emulated machine to be the default, to work better on lower spec host hardware.

That's to say though - what is improved on the A1200 version of Zeewolf?  If it's a speed thing, perhaps it is only needed to increase the CPU speed to 'fastest' rather than add AGA etc.

**Reply by sittingduck77** [reply]
I can understand the choice to target the 'minimum' emulated machine as it makes for the least taxing emulation which might mean that the game runs on even a Pi 2 (maybe?). I have a Pi 3B which runs most games at the right speed so 'upping' the hardware might start making the most taxing games worse. However, AFAIK whdload was made to run on pretty much any Amiga - and especially AGA Amigas. Is there a case to be made for using an 'A500' XML file for the Pis up to version 3, and an A1200 (or maybe even faster) XML file for faster Pis? Maybe Amiberry can be programmed to query the hardware? Especially flight sims and other vector based games could benefit quite a bit. Anyways, it is just a thought.

**Reply by midwan** [reply, marked as answer]
That's going to add more complexity, with little benefit.
I think the base hardware option (A600HD with extra RAM, Kickstart 2.05) works well for most cases that need an A500 config, and the A1200 one is selected anyway in case of AGA titles. That leaves a few special cases, which are non-AGA but would benefit from a faster processor (usually vector games, like Frontier). And those can be handled in the XML by changing the CPU model or speed setting.

Keeping the emulated hardware to the minimum required helps with the emulation speed, especially going forward with the newer "Preview" version of Amiberry, where the requirements are increased (due to higher emulation accuracy). If you're trying to emulate an 020 with AGA for an ECS game that doesn't need it, you're just wasting CPU power on your device.

**Reply by sittingduck77** [reply]
That makes sense, and I agree. Good discussion, though :-)

**Comment by giantclambake** [comment]
> Thanks for elaborating, giantclambake. Yeah, I'll set up individual configs for these games - I guess my point is, that the Whdload XML-file > should reflect that AGA setups are better for these games and run them that way pr. default. 

For sake of completeness, and although it's discouraged, you can directly edit the XML file itself ~ the problem with that, is anytime you update the XML file, the old XML file containing your edits is moved to a backup file, and you'd have to incorporate your edits into the newly downloaded XML file ....this is why it's discouraged ; not only does it risk the XML content being non-current, it opens the door to editing mistakes with the XML itself, plus you have to remember to move any edits across to the latest XML =)

That said (full disclosure ;), I do edit my XML directly, most of the time to remove every instance of `SCREEN_AUTOHEIGHT=TRUE`. That's because I can get a better display with that disabled...for all titles. I keep a backup of this modified XML. I don't consider asking anyone to change the XML, because SCREEN_AUTOHEIGHT=TRUE may be valid in many places ..ie; not x86-64 and not with a desktop sized monitor, and with earlier amiberry releases (I believe)...

Using config.uae files is therefore preferable, however, user must be aware these files exist, and that they override the XML.

Sometimes I think something like having a   `whdboot/game-data/user-whdload_db.txt ` file makes sense, which amiberry consults prior to parsing the XML, to see if there's a game-title match there first (with specific configurata). This would ensure any user edits to the (user) XML file are preserved and remain active/current across the release XML update (which will tend to stay pristine). This would avert the conf/ directory becoming cluttered with config.uae files, which invariably only get loaded/used when a whdload.lha is loaded/started.

**Reply by sittingduck77** [reply]
Good points, giantclambake - I like the idea of a whdload_db.txt file. But as long as I know how to make my own small changes, I'm happy :-)

**Reply by giantclambake** [reply]
I pointed out the real life example some time ago, with newly released Amiga titles being distributed in multiple formats, one of which being whdload.lha gamepack, and the installation instructions is one of user editing the main XML file... and for all the reasons above, that doesn't seem to be a 'permanent' installation step. This is not just one game either, many, and guess is it'll be ongoing.

It's subtly different to the previous usage example wherein you can create a config.uae file to control things, but this is only possible if an existing game snippet exists within the main XML file -- in these cases it doesn't ; user has to append the existing XML with the new game snippet, before they can create a config.uae that pertains to that XML game snippet. 

Personally, I think it's 'perilous' to let users near an otherwise 'automated' system file, that can be (or needs to be) updated at the push of a button...especially when the whole whdload feature of amiberry relies on the XML file. If there was a  user-whdload_db.txt file that was also scanned, which users could add or alter game XML snippets to, that's not so bad...ie; if user creates a badly formatted XML file, amiberry can fall-thru to the main XML ...and if no match is found, hope for the best =) The point is, the existing main XML file is less prone to edit damage...and they really shouldn't be telling ppl to edit it, but what else can you do? Wait for the required game XML snippet to be added to that repository, so one day when you press the update XML file button the whdload.lha works? 

That in a way, is the problem -- the only 'legitimate' way to get a (new) Amiga game added to the XML, is to have that game snippet added to the Horaceandthespider repo.

> But as long as I know how to make my own small changes

```
<game filename="Zeewolf_v5.03_0414" sha1="86e96aabae45bcb74aa2132884723863e50097c9">
                <name>Zeewolf</name>
                <subpath>Zeewolf</subpath>
                <variant_uuid>4ed5c4f5-c5f7-5d53-846a-dce26b9a4219</variant_uuid>
                <slave_count>1</slave_count>
                <slave_default>Zeewolf.slave</slave_default>
                <slave_libraries>False</slave_libraries>
                <slave number="1">
                        <filename>Zeewolf.slave</filename>
                        <datapath>data</datapath>
                </slave>
                <hardware>
                        CHIPSET=AGA
                        CLOCK=25
                        PRIMARY_CONTROL=MOUSE
                        PORT0=MOUSE
                        PORT1=JOY
                        SCREEN_AUTOHEIGHT=FALSE
                        SCREEN_HEIGHT=400
                        SCREEN_WIDTH=720
                        SCREEN_CENTERH=SMART
                        SCREEN_CENTERV=SMART
                </hardware>
        </game>
```

I included my config.uae file previously in example ~ to be fair, this examples the 'small changes' required to the XML, to achieve the same results... CHIPSET=AGA selects A1200/020, CLOCK sets cpu speed...

If a user-whdload_db.txt file existed, that's what you would do... just cut&paste the relevant XML snippet into the user-whdload_db.txt file, and edit that snippet ... right now, it doesn't, rock & a tight spot ;) ,,,

**Reply by midwan** [reply]
Why not just send the snippet to the upstream repo, to have it included there? Then everyone can benefit

**Reply by giantclambake** [reply]
None of my XML edits are of benefit to others.... unless they happen to use 4:3 monitors like my setup.
